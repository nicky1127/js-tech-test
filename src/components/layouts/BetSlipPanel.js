import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { connect } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import { Box, Typography } from "@material-ui/core";
import NoSelection from "components/units/NoSelection";
import Selections from "components/accordions/Selections";
import labels from "constants/labels";

const constants = labels.BetSlipPanel;

const useStyles = makeStyles(theme => ({
  root: {
    width: "23%",
    height: "100vh",
    position: "fixed",
    overflowY: "scroll",
    right: "0",
    backgroundColor: props =>
      props.selections.length > 0 ? "#fff" : "rgb(239,244,252)"
  },
  header: {
    boxSizing: "border-box",
    height: "10vh",
    padding: "0 8%",
    width: "100%",
    backgroundColor: "rgb(18,44,108)",
    display: "flex",
    alignItems: "center"
  },
  panelBody: {
    width: "100%",
    height: "85vh",
    border: props =>
      props.selections.length > 0 ? "1px solid rgb(165,175,194)" : "none"
  },
  numFlag: {
    width: "30px",
    height: "30px",
    color: "#fff",
    textAlign: "center",
    lineHeight: "30px",
    backgroundColor: "rgb(7,6,82)",
    borderRadius: "50%"
  },
  headerText: {
    color: "#fff",
    margin: " 0 15px"
  }
}));

export const BetSlipPanel = ({ children, selections }) => {
  const classes = useStyles({ selections });

  return (
    <div className={clsx("betSlipPanel", classes.root)}>
      <Box className={classes.header}>
        <Box className={classes.numFlag}>{selections.length}</Box>
        <Typography className={classes.headerText}>
          {constants.header}
        </Typography>
      </Box>
      <Box className={classes.panelBody}>
        {selections.length > 0 ? <Selections /> : <NoSelection />}
      </Box>
    </div>
  );
};

BetSlipPanel.defaultProps = {};

BetSlipPanel.propTypes = {};

const mapStateToProps = state => {
  const { selections } = state;
  return { selections };
};

const ConnectedBetSlipPanel = connect(mapStateToProps)(BetSlipPanel);

export default ConnectedBetSlipPanel;
