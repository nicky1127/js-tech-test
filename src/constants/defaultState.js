const state = {
  liveEvents: [],
  event: {},
  markets: {},
  isOddsDecimal: false,
  loading: false,
  error: {},
  selections: []
};

export default state;
