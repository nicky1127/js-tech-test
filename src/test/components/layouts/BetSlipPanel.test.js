import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import BetSlipPanel from "components/layouts/BetSlipPanel";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import defaultState from "constants/defaultState";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = {};

const state = {
  ...defaultState
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("BetSlipPanel", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BetSlipPanel {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("BetSlipPanel should render one [data-testid=betSlipPanel]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BetSlipPanel {...props} />
      </Provider>
    );
    const container = getByTestId("betSlipPanel");
    expect(container).toBeInTheDocument();
  });
});
