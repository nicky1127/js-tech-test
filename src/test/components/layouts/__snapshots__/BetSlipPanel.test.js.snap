// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetSlipPanel Snapshot matchs 1`] = `
<div
  className="betSlipPanel makeStyles-root-1 makeStyles-root-6"
  data-testid="betSlipPanel"
>
  <div
    className="MuiBox-root MuiBox-root-8 makeStyles-header-2"
  >
    <div
      className="MuiBox-root MuiBox-root-9 makeStyles-numFlag-4"
    >
      0
    </div>
    <p
      className="MuiTypography-root makeStyles-headerText-5 MuiTypography-body1"
    >
      My Bet Slip
    </p>
  </div>
  <div
    className="MuiBox-root MuiBox-root-10 makeStyles-panelBody-3 makeStyles-panelBody-7"
  >
    <div
      className="makeStyles-root-11"
      data-testid="noSelection"
    >
      <p
        className="MuiTypography-root makeStyles-noSelectText-12 MuiTypography-body1"
      >
        You have no selections in your Bet Slip
      </p>
      <div
        className="MuiBox-root MuiBox-root-14 makeStyles-optionBox-13"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Go to In-Play bets
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-15 makeStyles-optionBox-13"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Build an Accumulator
        </p>
      </div>
      <div
        className="MuiBox-root MuiBox-root-16 makeStyles-optionBox-13"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Next Races
        </p>
      </div>
    </div>
  </div>
</div>
`;
